netkan/netkan/mirrorer.py

```diff
@@ -313,12 +295,62 @@ def mirror_filename(self, with_epoch=True):
             Ckan.MIME_TO_EXTENSION[self.download_content_type],
         )
 
+    def mirror_source_filename(self, with_epoch=True):
+        return '{}-{}.source.zip'.format(
+            self.identifier,
+            self._format_version(with_epoch)
+        )
+
     def mirror_title(self, with_epoch=True):
         return '{} - {}'.format(
             self.name,
             self._format_version(with_epoch),
         )
 
+    @property
+    def item_metadata(self):
+        lic_urls = self.license_urls()
+        return {
+            'title':       self.mirror_title(),
+            'description': self.mirror_description,
+            'creator':     self.authors(),
+            'collection':  self.collection,
+            'subject':     'ksp; kerbal space program; mod',
+            'mediatype':   'software',
+            **({'licenseurl': lic_urls} if lic_urls else {}),
+        }
+
+    def open_download(self):
+        cached_file = self.cache_find_file
+        if cached_file:
+            # If the download is in the cache, use it
+            logging.info('Found matching cache entry at %s', cached_file)
+            return cached_file.open(mode='rb')
+        # Download the file as needed
+        with tempfile.NamedTemporaryFile() as tmp:
+            logging.info('Downloading %s', self.download)
+            urllib.request.urlretrieve(self.download, tmp.name)
+            # Copy to cache so the temp file can be deleted
+            shutil.copyfile(tmp.name, self.CACHE_PATH.joinpath(self.cache_filename))
+            logging.info('Downloaded %s to %s', self.mirror_item(), self.cache_filename)
+        return self.cache_find_file.open(mode='rb')
+
```

Comment:
I don't think we want to cache mirrored files, it's unlikely they'll be used by the inflator. But this should fix the current issue, so lets worry about that later.